File: typo-tester-typo-scala-old/src/scala/adventureworks/UpsertTwiceTest.scala
package adventureworks

import adventureworks.public.only_pk_columns.*
import org.junit.Assert.*
import org.junit.Test

class UpsertTwiceTest {
  private val onlyPkColumnsRepo = new OnlyPkColumnsRepoImpl

  @Test
  def secondUpsertShouldNotError(): Unit = {
    val row = OnlyPkColumnsRow("the answer is", 42)
    WithConnection {
      assertEquals(onlyPkColumnsRepo.upsert(row), onlyPkColumnsRepo.upsert(row))
    }
  }
}

File: typr-tester-typr-java/src/java/adventureworks/UpsertTwiceTest.java
package adventureworks;

import static org.junit.Assert.*;

import adventureworks.public_.only_pk_columns.OnlyPkColumnsRepoImpl;
import adventureworks.public_.only_pk_columns.OnlyPkColumnsRow;
import org.junit.Test;

public class UpsertTwiceTest {
  private final OnlyPkColumnsRepoImpl onlyPkColumnsRepo = new OnlyPkColumnsRepoImpl();

  @Test
  public void secondUpsertShouldNotError() {
    var row = new OnlyPkColumnsRow("the answer is", 42);
    WithConnection.run(
        c -> {
          assertEquals(onlyPkColumnsRepo.upsert(row, c), onlyPkColumnsRepo.upsert(row, c));
        });
  }
}
